src/main/java/dev/luzifer/log4jexploitfix/Log4JExploitFix.java

```diff
@@ -76,7 +79,14 @@ public void onPacketReceiving(PacketEvent event) {
                         packetContainer.getStrings().write(0, "");
     
                         logger.severe(player.getName() + " just tried something malicious");
-                        Bukkit.getPluginManager().callEvent(new PlayerMaliciousChatEvent(player, message));
+
+                        // call Event as sync
+                        new BukkitRunnable() {
+                            @Override
+                            public void run() {
+                                Bukkit.getPluginManager().callEvent(new PlayerMaliciousChatEvent(player, message));
+                            }
+                        }.runTask(plugin);
```

Comment:
```suggestion
                        }.runTask(Log4JExploitFix.this);
```

maybe even (depending on how @Luziferium prefers):
```java
Bukkit.getScheduler().runTask(Log4JExploitFix.this, () -> Bukkit.getPluginManager().callEvent(new PlayerMaliciousChatEvent(player, message)));
```